common/djangoapps/student/helpers.py

```diff
@@ -315,3 +315,14 @@ def get_next_url_for_login_page(request):
         # be saved in the session as part of the pipeline state. That URL will take priority
         # over this one.
     return redirect_to
+
+
+def get_email_domain (email):
+    email = email.lower().split('@')[1]
+    if 'gmail' in email:
+        spam_name = pgettext('gmail', 'Spam')
+    elif any(substring in email for substring in ['hotmail', 'outlook', 'live', 'outlook', 'msn']):
```

Comment:
Suffix the domains with `.` to be more specific.

`elif any(substring in '{}.'.format(email) for substring in ['hotmail', 'outlook', 'live', 'outlook', 'msn']):`

Also refactor this, it's a very long line in terms of ideas.